packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts

```diff
@@ -104,20 +109,62 @@ export const updateProfileHandler = async ({ ctx, input }: UpdateProfileOptions)
     data.emailVerified = null;
   }
 
-  const updatedUser = await prisma.user.update({
-    where: {
-      id: user.id,
-    },
-    data,
-    select: {
-      id: true,
-      username: true,
-      email: true,
-      metadata: true,
-      name: true,
-      createdDate: true,
-    },
-  });
+  let updatedUser;
+
+  // check if we are changing email and identity provider is not CAL
+  if (input.email && user.identityProvider !== IdentityProvider.CAL) {
+    const email = input.email;
+    // Only validate if we're changing email
+    if (email !== user.email) {
+      data.identityProvider = IdentityProvider.CAL;
+      data.identityProviderId = null;
+      updatedUser = await prisma.user.update({
+        where: {
+          id: user.id,
+        },
+        data,
+        select: {
+          id: true,
+          username: true,
+          email: true,
+          identityProvider: true,
+          identityProviderId: true,
+          metadata: true,
+          name: true,
+          createdDate: true,
+          locale: true,
+        },
+      });
+      if (updatedUser) {
+        // Because the email has changed, we are now attempting to use the CAL provider-
+        // which has no password yet. We have to send the reset password email.
+        await passwordResetRequest(updatedUser.email, updatedUser.locale || "en");
+        signOutUser = true;
+        passwordReset = true;
+      }
+    }
+  } else {
+    updatedUser = await prisma.user.update({
+      where: {
+        id: user.id,
+      },
+      data,
+      select: {
+        id: true,
+        username: true,
+        email: true,
+        metadata: true,
+        name: true,
+        createdDate: true,
+      },
+    });
```

Comment:
We can achieve the same without repeating this code. Let's update first and the handle post update on another condition.